python-package/lightgbm/engine.py

```diff
@@ -299,6 +328,90 @@ def handler_function(*args, **kwargs):
             return ret
         return handler_function
 
+    def __getstate__(self) -> Dict[str, Any]:
+        return vars(self)
+
+    def __setstate__(self, state: Dict[str, Any]) -> None:
+        vars(self).update(state)
+
+    def model_from_string(self, model_str: str) -> "CVBooster":
+        """Load CVBooster from a string.
+
+        Parameters
+        ----------
+        model_str : str
+            Model will be loaded from this string.
+
+        Returns
+        -------
+        self : CVBooster
+            Loaded CVBooster object.
+        """
+        self._from_dict(json.loads(model_str))
+        return self
+
+    def model_to_string(
+        self,
+        num_iteration: Optional[int] = None,
+        start_iteration: int = 0,
+        importance_type: str = 'split'
+    ) -> str:
+        """Save CVBooster to string.
```

Comment:
```suggestion
        """Save CVBooster to JSON string.
```

Can you please clarify that this method writes out JSON?

I totally understand (and support!) the decision to use JSON for this purpose, as that's a direction we'd like to move in this project (#4887).

But `CVBooster.model_to_string()` returning JSON does make it different from `Booster.model_to_string()`, which returns a LightGBM-custom-formatted text file.

https://github.com/microsoft/LightGBM/blob/c000b8cc689a9b3ee8bf4f294749d2f848f7176f/python-package/lightgbm/basic.py#L3401-L3408

Given that, I think it's important to over-communicate the fact that `CVBooster` uses JSON strings in its docs, since users may have been conditioned to think that the interface for `CVBooster` is very similar to the one for `Booster`.